### ISSUE TITLE
eventChannel emit END, unsubscribe not called

### ISSUE THREAD

**Issue by baldwmic** [redux-saga/redux-saga#875]
In the advanced docs for [using channels](https://github.com/redux-saga/redux-saga/blob/master/docs/advanced/Channels.md#using-the-eventchannel-factory-to-connect-to-external-events), I don't understand why the `clearInterval` is written in 2 places.

Here's the code for reference:
```
import { eventChannel, END } from 'redux-saga'

function countdown(secs) {
  return eventChannel(emitter => {
      const iv = setInterval(() => {
        secs -= 1
        if (secs > 0) {
          emitter(secs)
        } else {
          // this causes the channel to close
          emitter(END)
          clearInterval(iv)
        }
      }, 1000);
      // The subscriber must return an unsubscribe function
      return () => {
        clearInterval(iv)
      }
    }
  )
}
```

As I understand it, calling `emitter(END)` causes the channel to close. And according to the docs, the `unsubscribe` function is "`used by the channel to unsubscribe before the event source complete.`"

So, in the case of the above code, I would expect calling `emitter(END)` to cause the `unsubscribe` function to be executed since we just closed the channel by emitting the END event. However, this is not the case, as `clearInterval` is coded in both places, i.e. emitting END event does NOT call the unsubscribe function.

Is there something I'm not understanding about emitting an END event? Does the unsubscribe function only get called when the consumer saga calls `channel.close`?

**Comment by Andarist** [comment]
You are right, according to the source code the `unsubscribe` function will be called automaticallt when emiting `END`. Therefore doubling this `clearInterval` is futile. Would you like to send a PR with a docs change? Or should I update it myself?

Thanks for noticing!

**Comment by baldwmic** [comment]
I would like to provide a PR for this, though I may need some coaching since it will be my first 😄

**Comment by Andarist** [comment]
Well, basically you need to fork this repo from its main page on github. And then clone locally your version of it and create a new branch from the master. 

Once you push your changes out you will see a possibility to make a pull request on the repository's site.

Changes need to be made in the place I've linked in my previous comment. Would advise just to extract `close` function from the returning object and put it before `const unsubscribe`. Then just call this instead of `chan.close`.

**Comment by baldwmic** [comment]
Awesome, I will submit a PR by the end of the day! Thanks for walking me through the approach.

**Comment by Andarist** [comment]
You can also include in your commit's message something like "fixes #875", so this issue will be automatically closed when your PR gets merged in.

Also remember about updating the code snippet in the docs :) (the one which made you noticing this issue)
